codegen-cli/src/main/java/pro/bilous/codegen/process/ModelPropertyProcessor.kt

```diff
@@ -73,33 +86,113 @@ open class ModelPropertyProcessor(val codegen: CodeCodegen) {
+	private fun processIfListOrIdentityComplexModelEndsWithId(model: CodegenModel, property: CodegenProperty) {
 		if (property.isListContainer && property.datatypeWithEnum.startsWith("List")) {
 //			property.datatypeWithEnum = "Set" + property.datatypeWithEnum.removePrefix("List")
 			property.defaultValue = if (property.required) "listOf()" else "null"
 			model.imports.remove("List")
 			model.imports.remove("ArrayList")
-		} else if (property.isModel && !property.complexType.isNullOrEmpty() && property.complexType.endsWith("IdentityModel") && property.name.endsWith("Id")) {
+		} else if (property.isModel && !property.complexType.isNullOrEmpty() &&
+			property.complexType.endsWith("IdentityModel") &&
+			property.name.endsWith("Id")
+		) {
 			// convert Reference Type to the String
 			property.dataType = "String"
 			property.datatypeWithEnum = property.dataType
 			property.isModel = false
 			property.isString = true
 		}
-
-		if (isEnum(property)) {
-			convertToMetadataProperty(property, model)
-		}
-		addGuidAnnotation(property, model)
 	}
 
 	fun resolvePropertyType(property: CodegenProperty) {
 		if (property.vendorExtensions["columnType"] != null) {
 			return
 		}
+		val defaultStringSize = additionalProperties["defaultStringSize"]?.let { it as Int } ?: DEFAULT_STRING_SIZE
+		val databaseName = additionalProperties["database"]?.let { it as Database }?.name
+		when (databaseName) {
+			"mysql" -> {
+				resolvePropertyTypeForMySQL(property, defaultStringSize)
+			}
+			"postgresql" -> {
+				resolvePropertyTypeForPostgreSQL(property, defaultStringSize)
+			}
+			else -> resolvePropertyTypeByDefault(property, defaultStringSize)
+		}
+	}
+
+	private fun resolvePropertyTypeForMySQL(property: CodegenProperty, defaultStringSize: Int) {
+		when (property.datatypeWithEnum) {
+			"Boolean", "Boolean?" -> {
+				property.vendorExtensions["columnType"] = "\${BOOLEAN_VALUE}"
+				property.vendorExtensions["hibernateType"] = "java.lang.Boolean"
+				property.isBoolean = true
+			}
+			"Date", "Date?" -> {
+				property.vendorExtensions["columnType"] = "datetime"
+				property.vendorExtensions["hibernateType"] = "java.util.Date"
+				property.isDate = true
+			}
+			"Int", "Int?" -> {
+				property.vendorExtensions["columnType"] = "int"
+				property.isInteger = true
+			}
+			"BigDecimal", "BigDecimal?" -> {
+				property.vendorExtensions["columnType"] = "decimal(10,2)"
+				property.isNumber = true
+			}
+			"Long", "Long?" -> {
+				property.vendorExtensions["columnType"] = "bigint"
+				property.isNumber = true
+			}
+			else -> {
+				val maxLength =
+					if (property.maxLength != null && property.maxLength > 0) property.maxLength else defaultStringSize
+				property.vendorExtensions["columnType"] = "VARCHAR($maxLength)"
+				property.vendorExtensions["hibernateType"] = "java.lang.String"
+			}
+		}
+	}
+
+	private fun resolvePropertyTypeForPostgreSQL(property: CodegenProperty, defaultStringSize: Int) {
+		when (property.datatypeWithEnum) {
```

Comment:
@agentfan what the reason for the duplication of all property type resolvers for postgres and mysql?